cli/internal/cache/cache.go

```diff
@@ -271,31 +300,34 @@ func (mplex *cacheMultiplexer) Fetch(anchor turbopath.AbsoluteSystemPath, key st
 			// the operation. Future work that plumbs UI / Logging into the cache system
 			// should probably log this at least.
 		}
+
 		if ok {
 			// Store this into other caches. We can ignore errors here because we know
 			// we have previously successfully stored in a higher-priority cache, and so the overall
 			// result is a success at fetching. Storing in lower-priority caches is an optimization.
-			_ = mplex.storeUntil(anchor, key, duration, actualFiles, i)
+			_ = mplex.storeUntil(anchor, key, itemStatus.TimeSaved, actualFiles, i)
 
 			// If another cache had already set this to true, we don't need to set it again from this cache
-			combinedCacheState.Local = combinedCacheState.Local || itemStatus.Local
-			combinedCacheState.Remote = combinedCacheState.Remote || itemStatus.Remote
-			return combinedCacheState, actualFiles, duration, err
+			combinedCacheState.Hit = itemStatus.Hit
+			combinedCacheState.Source = itemStatus.Source
+			combinedCacheState.TimeSaved = itemStatus.TimeSaved
+			return combinedCacheState, actualFiles, nil
```

Comment:
This logic change is a bit concerning. Previously, the `||` logic resulted in "first cache hit wins". Now it's "last one wins". But because there's a return statement, it's still "first one wins". So I _think_ we're effectively at the same result, but I could use another pair of eyes to confirm that I'm thinking about this right.